src/Splat.Microsoft.Extensions.DependencyInjection/MicrosoftDependencyResolver.cs

```diff
@@ -0,0 +1,213 @@
+        private IServiceCollection _serviceCollection;
+        private bool _isImmutable;
+        private IServiceProvider _serviceProvider;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MicrosoftDependencyResolver" /> class with an <see cref="IServiceCollection"/>.
+        /// </summary>
+        /// <param name="services">An instance of <see cref="IServiceCollection"/>.</param>
+        public MicrosoftDependencyResolver(IServiceCollection services = null)
+        {
+            _serviceCollection = services ?? new ServiceCollection();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MicrosoftDependencyResolver" /> class with a configured service Provider.
+        /// </summary>
+        /// <param name="serviceProvider">A ready to use service provider.</param>
+        public MicrosoftDependencyResolver(IServiceProvider serviceProvider)
+        {
+            if (serviceProvider == null)
+            {
+                throw new ArgumentNullException(nameof(serviceProvider));
+            }
+
+            _serviceProvider = serviceProvider;
+            _isImmutable = true;
+        }
+
+        /// <inheritdoc />
+        public virtual object GetService(Type serviceType, string contract = null)
+        {
+            if (serviceType == null)
+            {
+                throw new ArgumentNullException(nameof(serviceType));
+            }
+
+            var sp = GetServiceProvider(contract);
+            return sp.GetRequiredService(serviceType);
+        }
+
+        /// <inheritdoc />
+        public virtual IEnumerable<object> GetServices(Type serviceType, string contract = null)
+        {
+            if (serviceType == null)
+            {
+                throw new ArgumentNullException(nameof(serviceType));
+            }
+
+            var sp = GetServiceProvider(contract);
+            return sp.GetServices(serviceType);
+        }
+
+        /// <inheritdoc />
+        public virtual void Register(Func<object> factory, Type serviceType, string contract = null)
+        {
+            if (_isImmutable)
+            {
+                throw new InvalidOperationException(ImmutableExceptionMessage);
+            }
+
+            if (serviceType == null)
+            {
+                throw new ArgumentNullException(nameof(serviceType));
+            }
+
+            if (contract != null)
+            {
+                _serviceCollection.AddScoped(serviceType, _ => factory());
+            }
+            else
+            {
+                _serviceCollection.AddTransient(serviceType, _ => factory());
+            }
+        }
+
+        /// <inheritdoc/>
+        public void UnregisterCurrent(Type serviceType, string contract = null)
+        {
+            throw new NotImplementedException();
```

Comment:
You can most likely do unregister similar to how david did it for the other read only collections. Eg re-create the ServiceCollection from the existing members minus the one that wants to be removed. Bit of worked involved with that so probably a second pass feature.

Reply:
Add an issue for that though.

Reply:
I think it's a bad idea. Because this is assuming I'm having my DI container maintained by Splat, but what if I use [.NET Generic Host](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/generic-host?view=aspnetcore-2.2) or some other facility to maintain my DI container?

Reply:
Some users do, and every other container regardless provide this functionality. You are potentially breaking users unless we do.

Reply:
I mean if users choose to use MSDI, they're probably willing to play by MSDI's rules no?
See our discussion here: https://github.com/reactiveui/splat/pull/370#discussion_r309724017

Reply:
I mean if users choose to use MSDI, they're probably willing to play by MSDI's rules no?
See our discussion here: https://github.com/reactiveui/splat/issues/378#issuecomment-518467934.